src/main/java/org/folio/circulation/domain/Item.java

```diff
@@ -423,4 +400,14 @@ public Item withLoanType(JsonObject newLoanTypeRepresentation) {
       this.primaryServicePoint,
       newLoanTypeRepresentation);
   }
+
+  private CallNumberComponents fetchCallNumberComponents(JsonObject itemRepresentation) {
```

Comment:
I'd prefer that the call number components were a parameter to the item constructor, and did not require interpretation of the JSON within the class.

This is because there is a general intent to move away from the hidden JSON representations in the domain, as it adds complexity, mixes concerns and creates coupling between the API, storage API and the domain.